cli/module_registry.go

```diff
@@ -130,76 +128,48 @@ func UpdateModuleCommand(c *cli.Context) error {
 		return err
 	}
 
-	moduleID, err := parseModuleID(manifest.Name)
+	moduleID, err := resolveModuleIDFromMultipleSources(c, client, manifest.Name, publicNamespaceArg, orgIDArg)
 	if err != nil {
 		return err
 	}
-	// TODO(APP-2230) This logic is duplicated in update and upload and should be refactored
-	// TODO(APP-2230) Simplify this once we can pass org-id in ModuleID
-	if publicNamespaceArg != "" {
-		switch moduleID.Prefix {
-		case "":
-			moduleID.Prefix = publicNamespaceArg
-		case publicNamespaceArg:
-			// the meta.json manifest == public_namespace arg
-			fmt.Fprint(c.App.Writer, "The module's meta.json already specifies a public namespace. Ignoring\n")
-		default:
-			// the meta.json manifest != public_namespace arg
-			// we may want to investigate a better way of handling this error case
-			// For now, it seems like bad UX to ignore this error
-			return errors.Errorf("the module's meta.json specifies a namespace of '%s'"+
-				" but a namespace of '%s' was provided in command line arguments",
-				moduleID.Prefix, publicNamespaceArg)
-		}
-	}
-	var orgID *string
-	if orgIDArg != "" {
-		if moduleID.Prefix == "" {
-			orgID = &orgIDArg
-		} else {
-			fmt.Fprintf(c.App.Writer, "A public namespace (%s) is specified in the config."+
-				" It is not necessary to provide an org_id\n", moduleID.Prefix)
-		}
-	}
-	if orgID == nil && moduleID.Prefix == "" {
-		return errors.New("The module's namespace is not set in the meta.json." +
-			" You must provide a public_namespace (if you have set one) or supply your org id")
-	}
-	manifest.Name = moduleID.toString()
-	// end duplicated logic
 
-	response, err := client.UpdateModule(manifest, orgID)
-	fmt.Fprintf(c.App.Writer, "Module successfully updated! You can view your changes online here: %s\n", response.GetUrl())
+	response, err := client.UpdateModule(moduleID, manifest)
 	if err != nil {
 		return err
 	}
+	fmt.Fprintf(c.App.Writer, "Module successfully updated! You can view your changes online here: %s\n", response.GetUrl())
 
 	// If the namespace isn't set, modify the meta.json to set it (if available)
-	if moduleID.Prefix == "" {
-		org, err := resolveOrg(client, orgIDArg, publicNamespaceArg)
+	manifestModuleID, err := parseModuleID(manifest.Name)
+	if err != nil {
+		return err // shouldn't happen because this has already been parsed
+	}
+	if manifestModuleID.Prefix == "" || isValidOrgID(manifestModuleID.Prefix) {
+		org, err := getOrgByModuleIDPrefix(client, moduleID.Prefix)
 		if err != nil {
-			return err
+			// hopefully a user never sees this. An alternative would be to fail silently here
+			// to prevent the user from being surprised/scared that their update failed
+			return errors.Wrap(err, "error while trying to tidy up the local meta.json")
```

Comment:
This err msg seems off... I am trying indicate to a user that not everything succeeded, but that their intended action succeeded 

Could also fmt.Fprintf(app.ErrWriter, err.Error()) this so that the command still returns exit status 0 but so that any human will see the error

Reply:
Above we are already printing `Module successfully updated!...` So I think the user will get that the module update happened and we can just say `Failed to update meta.json with new information from Viam.` This is an edge case that will only happen if App and CLI are not compatible with each other, right?

Reply:
That is a much better error message. Switching to that.

Yup, I suspect this error case is "unrechable" but should be handled anyway